### ISSUE TITLE
Bitfinex returns scientific notation in its direct API - number: String has no effect

### ISSUE THREAD

**Issue by quakemmo** [ccxt/ccxt#24135]
### Operating System

Linux

### CCXT Version

latest

### Description

The exchange({'number': String}) seems to work, EXCEPT for the bitfinex2 exchange. Still returns scientific notations for OHLCV, for example for BTC/EUR (can be very small volumes for short candles).

After checking the Bitfinex v2 API directly on their servers, they actually serve a scientific notated float in these cases.

CCXT's conversion to String doesn't seem to have any effect on that.

I'm obviously using a hack to convert stuff on the fly, but this doesn't seem like something we should do as a hack, considering we're working with money :)

Or maybe is there some option for CCXT to do it itself? As it is now, I get, from CCXT's fetchOHLCV(), proper numeric strings from all exchanges except bitfinex2, where I occasionally get ugly scientifc notation that gives my computer migraines.

### Code

```
  

```

**Comment by carlosmiei** [comment]
Hello @quakemmo,

Thanks for reporting it. Does it only happen with `fetchOHLCV` ?

**Comment by carlosmiei** [comment]
@quakemmo I tested it and didn't get the scientific notation, can you please show me the code you're running?

**Comment by quakemmo** [comment]
Run from the examples/js ccxt directory.
Look at the first line of the output.

```
#!/usr/bin/node
'use strict';
import ccxt from '../../js/ccxt.js';

const cex = new ccxt.bitfinex2({'number': String});
await cex.loadMarkets();

const since = 1730547300000;
let ret = await cex.fetchOHLCV('BTC/EUR', '5m', since);

console.log(ret);

```

**Comment by carlosmiei** [comment]
@quakemmo Sorry I was testing the python version, I will recheck it and get back to you

**Comment by quakemmo** [comment]
Also I would like to add, but that's maybe/probably not relevant to this issue (but in case someone from bitfinex reads this) - the API reports things like this:

```
 [
    1730550300000,
    '63814',
    '63814',
    '63796',
    '63796',
    '0.00015000000000000001'
  ],
```

I don't know if this is considered OK. Sure looks ugly to me.

**Comment by carlosmiei** [comment]
@quakemmo Yeah regarding the raw response itself we can't do anything about it :/

**Comment by carlosmiei** [comment]
@quakemmo 
Can you please upgrade your ccxt version and try again? it should be working now

**Comment by quakemmo** [comment]
I will test and report here later. After seeing your patch, I see you handle the scientific notation for small numbers, but not the big ones. Maybe it should be there as well.

**Comment by carlosmiei** [comment]
@quakemmo You're right I can check `e` instead of `e-` but do you know if the same is happening with larger numbers? Usually they're not that common in this context

**Comment by kroitor** [comment]
Hello @quakemmo 👋

Closing this during a cleanup — verified against the current version: both directions of scientific notation are now expanded by ccxt's number-to-string conversion. Your original small-number case (`1.5e-7` → `0.00000015`) was fixed back in November 2024, and the large-number direction you flagged afterwards is handled too (`1.5e+21` → `1500000000000000000000`), so `{'number': String}` on bitfinex2 returns plain decimal strings for OHLCV regardless of how Bitfinex formats the raw response.

Thanks for the report and for catching the second case — both are covered. 🙏
